web/server/entrypoint.go

```diff
@@ -398,7 +471,22 @@ func (s *robotServer) serveWeb(ctx context.Context, cfg *config.Config) (err err
 		robotOptions = append(robotOptions, robotimpl.WithFTDC())
 	}
 
-	myRobot, err := robotimpl.New(ctx, processedConfig, s.logger, robotOptions...)
+	// Create `minimalProcessedConfig`, a copy of `fullProcessedConfig`. Remove
+	// all components, services, remotes, modules, and processes from
+	// `minimalProcessedConfig`. Create new robot with `minimalProcessedConfig`
+	// and immediately start web service. We need the machine to be reachable
+	// through the web service ASAP, even if some resources take a long time to
+	// initially configure.
+	minimalProcessedConfig := *fullProcessedConfig
+	minimalProcessedConfig.Components = nil
+	minimalProcessedConfig.Services = nil
+	minimalProcessedConfig.Remotes = nil
+	minimalProcessedConfig.Modules = nil
+	minimalProcessedConfig.Processes = nil
+
+	// Start robot in an initializing state with minimal config.
+	robotOptions = append(robotOptions, robotimpl.WithInitializing())
```

Comment:
What's the consequence of not having this line?
* Do we go back to the old behavior of having to do an initial reconfigure before opening a server port?
* Or do we still open a server port before configuring -- but we just lie to clients that the robot is in "MachineState.RUNNING"?

Reply:
> Or do we still open a server port before configuring -- but we just lie to clients that the robot is in "MachineState.RUNNING"?

This one is the consequence; the robot would falsely report itself as running.